exonum-java-binding-core/rust/src/utils/jni_cache.rs

```diff
@@ -0,0 +1,158 @@
+use jni::{
+    objects::{GlobalRef, JMethodID},
+    sys::{jint, JNI_VERSION_1_8},
+    JNIEnv, JavaVM,
+};
+use std::os::raw::c_void;
+
+static mut OBJECT_CACHED_GET_CLASS: Option<JMethodID> = None;
+static mut CLASS_CACHED_GET_NAME: Option<JMethodID> = None;
+static mut THROWABLE_CACHED_GET_MESSAGE: Option<JMethodID> = None;
+
+static mut UTA_CACHED_EXECUTE: Option<JMethodID> = None;
+static mut UTA_CACHED_INFO: Option<JMethodID> = None;
+static mut UTA_CACHED_VERIFY: Option<JMethodID> = None;
+
+static mut USA_CACHED_STATE_HASHES: Option<JMethodID> = None;
```

Comment:
Cached USA states - sounds quite funny. Although names of these statics and method names are not much readable, unfortunately. We are not using such abbreviations commonly, but I don't know how we can avoid them without blowing method names lengths.

Reply:
Statics are not exposed to the public so I believe it's OK. And I make method names short on purpose because of initial intention of using them prefixed with corresponding module's part. Like `jni_cache::get_error_class()`. This way it's kinda more clear that cached `error_class` comes from `jni_cache` entity. And this from my POV gives us an ability to just import the `jni_cache` in client classes and write prefixed methods instead of importing a bunch of explicit methods like `cache_get_uta_cached_execute`. So this was a reason I asked you if this approach is better or this is more clear to me only.

Reply:
I'd consider an option of grouping constants related to (some) classes together in a namespace, please see below. Don't know if it can be achieved easily :man_shrugging: